expenses-manager-frontend/src/main/frontend/src/app/expense-table/expense-table.component.html

```diff
@@ -0,0 +1,110 @@
+<div class="datepicker-wrapper">
+  <div class="datepicker-buttons">
+    <button class="date-buttons" (click)="goToPreviousMonth()">&lt;</button>
+    <button class="date-buttons date-picker-button" (click)="openDatepicker(datepicker)">
+      <input matInput [matDatepicker]="datepicker" class="hidden-datepicker-input">
+      {{ displayMonthAndYear() }} <mat-icon>expand_more</mat-icon>
+    </button>
+    <button class="date-buttons" (click)="goToNextMonth()">&gt;</button>
+  </div>
+  <mat-datepicker #datepicker
+                  startView="multi-year"
+                  (monthSelected)="setMonthAndYear($event, datepicker)">
+  </mat-datepicker>
+</div>
+
+<table mat-table [dataSource]="expenses" matSort class="mat-elevation-z8 expense-table">
+  
+  <ng-container matColumnDef="actions">
+    <th mat-header-cell *matHeaderCellDef class="column"> Actions </th>
+    <td mat-cell class="column" *matCellDef="let element"><button class="actions-button"><mat-icon>more_vert</mat-icon></button> </td>
+  </ng-container>
+  
+  <ng-container matColumnDef="date">
+    <th mat-header-cell *matHeaderCellDef mat-sort-header class="column"> Date </th>
+    <td mat-cell class="column" *matCellDef="let element"> {{element.date[2] | number : '2.0-0'}}/{{element.date[1] | number : '2.0-0'}}/{{element.date[0]}} </td>
+  </ng-container>
+  
+  <ng-container matColumnDef="amount">
+    <th mat-header-cell *matHeaderCellDef mat-sort-header class="column"> Amount </th>
+    <td mat-cell class="column" *matCellDef="let element"> {{element.amount}} </td>
+  </ng-container>
+  
+  <ng-container matColumnDef="currency">
+    <th mat-header-cell *matHeaderCellDef class="column">
+      <div class="filter-header"> Currency 
+        <button mat-icon-button class="filter-button">
+          <img src="../../assets/images/filter.svg">
+        </button>
+      </div>
+    </th>
+    <td mat-cell class="column" *matCellDef="let element"> {{getCurrencyCode(element.currencyId)}} </td>
+  </ng-container>
+
+  <ng-container matColumnDef="category">
+    <th mat-header-cell *matHeaderCellDef class="column">
+      <div class="filter-header"> Category
+        <button mat-icon-button class="filter-button">
+          <img src="../../assets/images/filter.svg">
+        </button>
+      </div>
+    </th>
+    <td mat-cell class="column" *matCellDef="let element">
+      <div class="category-text" [ngStyle]="{'color': getCategoryColor(element.categoryId), 
+      'background-color': hexToRgba(getCategoryColor(element.categoryId))}">
```

Comment:
Aici inteleg ca vrei sa setezi o transparenta a culorii. Dar nu cred ca avem nevoie de o functie pentru asta. Putem seta din css transparenta aplicand opacity: 0.2 pe clasa category-text si pastrand codul in hex cum vine din backend.